numba/tests/test_dyn_array.py

```diff
@@ -0,0 +1,362 @@
+        arr_a = np.random.random(10).reshape(2, 5)
+        arr_b = np.random.random(10).reshape(2, 5)
+
+        np.testing.assert_equal(pyfunc(arr_a, arr_b),
+                                cfunc(arr_a, arr_b))
+
+        # 3D case
+        arr_a = np.random.random(70).reshape(2, 5, 7)
+        arr_b = np.random.random(70).reshape(2, 5, 7)
+
+        np.testing.assert_equal(pyfunc(arr_a, arr_b),
+                                cfunc(arr_a, arr_b))
+
+    def test_multithread(self):
+
+        def pyfunc(inp):
+            out = np.empty(inp.size)
+            tmp = 0
+            for i in range(out.size):
+                out[i] = tmp
+                tmp = inp[i]
+            return out
+
+        cfunc = nrtjit(pyfunc)
+        size = 10**5
+        arr = np.random.randint(0, 1, size)
+
+        np.testing.assert_equal(pyfunc(arr), cfunc(arr))
+
+        workers = []
+        inputs = []
+        outputs = []
+
+        # Make wrapper to store the output
+        def wrapped(inp, out):
+            out[:] = cfunc(inp)
+
+        # Create worker threads
+        for i in range(10):
+            arr = np.random.randint(0, 1, size)
+            out = np.empty_like(arr)
+            thread = threading.Thread(target=wrapped,
+                                      args=(arr, out),
+                                      name="worker{0}".format(i))
+            workers.append(thread)
+            inputs.append(arr)
+            outputs.append(out)
+
+        # Launch worker threads
+        for thread in workers:
+            thread.start()
+
+        # Join worker threads
+        for thread in workers:
+            thread.join()
+
+        # Check result
+        for inp, out in zip(inputs, outputs):
+            np.testing.assert_equal(pyfunc(inp), out)
+
+    def test_multithread_stress(self):
+
+        def pyfunc(n, t):
+            out = np.empty(n)
+            for i in range(out.size):
+                out[i] = i
+
+            for i in range(t):
+                tmp = np.empty(n)
+                for j in range(tmp.size):
+                    tmp[j] = i + j
+                for j in range(out.size):
+                    tmp[j] += out[j]
+                out = tmp   # Swap the array here
+
+            return out
+
+
+        cfunc = nrtjit(pyfunc)
+        size = 1000
+        repeat = 2000
+
+        expected = pyfunc(size, repeat)
+        np.testing.assert_equal(expected, cfunc(size, repeat))
+
+        workers = []
+        outputs = []
+
+        # Make wrapper to store the output
+        def wrapped(n, t, out):
+            out[:] = cfunc(n, t)
+
+        # Create worker threads
+        for i in range(10):
+            out = np.empty(size)
+            thread = threading.Thread(target=wrapped,
+                                      args=(size, repeat, out),
+                                      name="worker{0}".format(i))
+            workers.append(thread)
+            outputs.append(out)
+
+        # Launch worker threads
```

Comment:
Similarly, I'm not sure what you're testing here. As long as the threads are serialized and don't access the same objects, there's not much to test for :-) (the freelist perhaps?)